pkg/tests/api/alerting/api_notification_channel_test.go

```diff
@@ -0,0 +1,956 @@
+			  "title": "Details"
+			}
+		  ],
+		  "summary": "[firing:1]  (TeamsAlert)",
+		  "themeColor": "#D63232",
+		  "title": "[firing:1]  (TeamsAlert)"
+		}`,
+	},
+	"webhook_recv/webhook_test": {
+		`{
+		  "receiver": "webhook_recv",
+		  "status": "firing",
+		  "alerts": [
+			{
+			  "status": "firing",
+			  "labels": {
+				"alertname": "WebhookAlert"
+			  },
+			  "annotations": {},
+			  "startsAt": "%s",
+			  "endsAt": "0001-01-01T00:00:00Z",
+			  "generatorURL": "",
+			  "fingerprint": "a0e2a99fc56fce63"
+			}
+		  ],
+		  "groupLabels": {},
+		  "commonLabels": {
+			"alertname": "WebhookAlert"
+		  },
+		  "commonAnnotations": {},
+		  "externalURL": "http://localhost:3000/",
+		  "version": "1",
+		  "groupKey": "{}/{alertname=\"WebhookAlert\"}:{}",
+		  "truncatedAlerts": 0,
+		  "title": "[FIRING:1]  (WebhookAlert)",
+		  "state": "alerting",
+		  "message": "\n**Firing**\nLabels:\n - alertname = WebhookAlert\nAnnotations:\nSource: \n\n\n\n\n"
+		}`,
+	},
+}
+
+func getAlertmanagerConfig(channelAddr string) string {
+	return strings.ReplaceAll(alertmanagerConfig, "CHANNEL_ADDR", channelAddr)
+}
+
+func getExpAlertmanagerConfigFromAPI(channelAddr string) string {
+	return strings.ReplaceAll(expAlertmanagerConfigFromAPI, "CHANNEL_ADDR", channelAddr)
+}
+
+// alertNames are name of alerts to be sent. This should be in sync with
+// the routes that we define in Alertmanager config.
+// EmailAlert and TelegramAlert are missing because they don't
+// send a JSON. Email and POST body are yet to be supported in the tests.
+var alertNames = []string{"DingDingAlert", "SlackAlert1", "SlackAlert2", "PagerdutyAlert", "TeamsAlert", "WebhookAlert"}
+
+func getRulesConfig(t *testing.T) string {
+	interval, err := model.ParseDuration("10s")
+	require.NoError(t, err)
+	rules := apimodels.PostableRuleGroupConfig{
+		Name:     "arulegroup",
+		Interval: interval,
+	}
+
+	// Create rules that will fire as quickly as possible for all the routes.
+	for _, alertName := range alertNames {
+		rules.Rules = append(rules.Rules, apimodels.PostableExtendedRuleNode{
+			GrafanaManagedAlert: &apimodels.PostableGrafanaRule{
+				Title:     alertName,
+				Condition: "A",
+				Data: []ngmodels.AlertQuery{
+					{
+						RefID: "A",
+						RelativeTimeRange: ngmodels.RelativeTimeRange{
+							From: ngmodels.Duration(time.Duration(5) * time.Hour),
+							To:   ngmodels.Duration(time.Duration(3) * time.Hour),
+						},
+						DatasourceUID: "-100",
+						Model: json.RawMessage(`{
+							"type": "math",
+							"expression": "2 + 3 > 1"
+						}`),
+					},
+				},
+			},
+		})
+	}
+
+	b, err := json.Marshal(rules)
+	require.NoError(t, err)
+
+	return string(b)
+}
+
+func TestNotificationChannels(t *testing.T) {
+	dir, path := testinfra.CreateGrafDir(t, testinfra.GrafanaOpts{
+		EnableFeatureToggles: []string{"ngalert"},
+		AnonymousUserRole:    models.ROLE_EDITOR,
```

Comment:
```suggestion
		DisableAnonymous: true,
```
In #33283 I have changed `createFolder()` (called below) and the folder is created with the main organisation instead of the anonymous organisation.
Therefore we have to disable the anonymous mode, [create a user](https://github.com/grafana/grafana/blob/b971d570fa17589388f2878cecffdf7a93a0d78f/pkg/tests/api/alerting/api_prometheus_test.go#L36) and [make authenticated requests](https://github.com/grafana/grafana/blob/b971d570fa17589388f2878cecffdf7a93a0d78f/pkg/tests/api/alerting/api_prometheus_test.go#L57).

Reply:
Perfect, thanks for this! I was wondering why it was giving 404 in CI and this should fix it.